ansible/roles/operator-pipeline/templates/openshift/tasks/preflight-trigger.yml

```diff
@@ -46,188 +46,166 @@ spec:
+            exit 0
+        fi
 
-          if [ "$(params.skip)" == "true" ]; then
-              echo "Preflight testing has been skipped"
-              exit 0
-          fi
+        set -e
 
-          echo "Cloning https://github.com/openshift/release"
-          git clone https://github.com/openshift/release
+        if [ -f "$(workspaces.credentials.path)/project_dockerconfigjson" ]
+        then
+            preflight-trigger encrypt --gpg-encryption-private-key "${PIPELINE_GPG_DECRYPTION_PRIVATE_KEY}" --gpg-encryption-public-key "${PIPELINE_GPG_DECRYPTION_PUBLIC_KEY}" --pflt-docker-config "$(workspaces.credentials.path)/project_dockerconfigjson" --output-path "$(workspace.credentials.path)/project_dockerconfigjson.gpg"
+            preflight-trigger encode --file "$(workspace.credentials.path)/project_dockerconfigjson.gpg" --output-path "$(workspace.credentials.path)/project_dockerconfigjson.gpg.b64"
+        fi
+      volumeMounts:
+        - mountPath: /etc/gpg-key
+          name: gpg-key-volume
+          readOnly: true
       workingDir: $(workspaces.output.path)
     - env:
         - name: KUBECONFIG
           value: /etc/kubeconfig-volume/$(params.prow_kubeconfig_secret_key)
-      image: "$(params.preflight_trigger_image)"
+      image: $(params.preflight_trigger_image)
       name: create-openshift-ci-prowjob
       resources: {}
+      script: >
+        #!/usr/bin/env sh
+
+        if [ "$(params.skip)" == "true" ]
+        then
+            echo "Preflight testing has been skipped"
+            exit 0
+        fi
+
+        DPTP_VER="4.9"
+        CP_VER="$(params.ocp_version)"
+
+        if [ "$(printf '%s\n' "$DPTP_VER" "$CP_VER" | sort -V | head -n1)" = "$DPTP_VER" ]
+        then
+            OCP_VER="${DPTP_VER}"
+        else
+            OCP_VER="${CP_VER}"
+        fi
+
+
+        if [ "$(params.ocp_version)" == "4.6" ] || [ "$(params.ocp_version)" == "4.7" ]
+        then
+            JOB_SUFFIX="aws"
+        else
+            JOB_SUFFIX="claim"
+        fi
+
+        set -e
+
+        if [ -f "$(workspaces.credentials.path)/project_dockerconfigjson.gpg.b64" ]
+        then
+            PFLT_DOCKERCONFIG="$(cat "$(workspaces.credentials.path)/project_dockerconfigjson.gpg.b64")"
+            export PFLT_DOCKERCONFIG
+
+            preflight-trigger \
+        -preprod \
+        -ocp-version "${OCP_VER}" \
+        -pflt-docker-config "${PFLT_DOCKERCONFIG}" \
+        -pflt-log-level "$(params.log_level)" \
+        -pflt-index-image "$(params.bundle_index_image)" \
+        -test-asset "$(params.bundle_image)" \
+        -asset-type "$(params.asset_type)"
+        else
+            preflight-trigger \
+        -preprod \
+        -ocp-version "${OCP_VER}" \
+        -pflt-log-level "$(params.log_level)" \
+        -pflt-index-image "$(params.bundle_index_image)" \
+        -test-asset "$(params.bundle_image)" \
+        -asset-type "$(params.asset_type)"
+        fi
```

Comment:
Relatively unsure of how this spacing plays out in this nested YAML-y setup.

Reply:
Existing task uses the same spacing.